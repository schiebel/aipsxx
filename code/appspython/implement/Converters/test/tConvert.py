import numarray as NUM

from _tConvert import *

t = tConvert();

print t.testbool (True);
print t.testbool (False);
print t.testint (-1);
print t.testint (10);
print t.testfloat (3.14);
print t.testfloat (12);
print t.teststring ("this is a string");

print t.testvecint ([1,2,3,4]);
print t.testveccomplex ([1+2j, -1-3j, -1.5+2.5j]);
print t.testvecstr (["a1","a2","b1","b2"])
print t.testipos ([2,3,4]);

print t.testvh (True);
print t.testvh (2);
print t.testvh (1.3);
print t.testvh (10-11j);
print t.testvh ("str");
print t.testvh ([True]);
print t.testvh ([2,4,6,8,10]);
print t.testvh ([1.3,4,5,6]);
print t.testvh ([10-11j,1+2j]);
print t.testvh (["str1","str2"]);
print t.testvh ({"shape":[2,2],"array":["str1","str2","str3","str4"]});
a  =  t.testvh ({"shape":[2,2],"array":["str1","str2","str3","str4"]});
print a;
print t.testvh (a);

a  =  t.testvh ([10-11j,1+2j]);
print a.shape;
print a.type();
print t.testvh (a);

b  =  NUM.array([[2,3],[4,5]]);
print b;
print t.testvh (b);
