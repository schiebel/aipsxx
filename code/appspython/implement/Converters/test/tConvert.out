bool 1
True
bool 0
False
Int -1
-1
Int 10
10
Float 3.14
3.1400001049
Float 12
12.0
String this is a string
this is a string
VecInt [1, 2, 3, 4]
(1, 2, 3, 4)
VecComplex [(1,2), (-1,-3), (-1.5,2.5)]
((1+2j), (-1-3j), (-1.5+2.5j))
VecStr [a1, a2, b1, b2]
('a1', 'a2', 'b1', 'b2')
IPos [2, 3, 4]
(2, 3, 4)
VH Bool
True
VH Int
2
VH double
1.3
VH DComplex
(10-11j)
VH String
str
VH Array<Bool>
[1]
VH Array<Int>
[ 2  4  6  8 10]
VH Array<double>
[ 1.3  4.   5.   6. ]
VH Array<DComplex>
[ 10.-11.j   1. +2.j]
VH Array<String>
('str1', 'str2')
VH Array<String>
{'shape': (2, 2), 'array': ('str1', 'str2', 'str3', 'str4')}
VH Array<String>
{'shape': (2, 2), 'array': ('str1', 'str2', 'str3', 'str4')}
VH Array<String>
{'shape': (2, 2), 'array': ('str1', 'str2', 'str3', 'str4')}
VH Array<DComplex>
(2,)
Complex64
VH Array<DComplex>
[ 10.-11.j   1. +2.j]
[[2 3]
 [4 5]]
VH Array<Int>
[[2 3]
 [4 5]]
